kedro/extras/datasets/datatable/csv_dataset.py

```diff
@@ -0,0 +1,178 @@
+        on a specific filesystem.
+
+        Args:
+            filepath: Filepath in POSIX format to a CSV file prefixed with a protocol like `s3://`.
+                If prefix is not provided, `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            load_args: Datatable options for loading CSV files.
+                Here you can find all available arguments:
+                https://datatable.readthedocs.io/en/latest/api/dt/fread.html
+                All defaults are preserved.
+            save_args: Datatable options for saving CSV files.
+                Saving is done via pandas.
+                Here you can find all available arguments:
+                https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_csv.html
+                All defaults are preserved, but "index", which is set to False.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``), as well as
+                to pass to the filesystem's `open` method through nested keys
+                `open_args_load` and `open_args_save`.
+                Here you can find all available arguments for `open`:
+                https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
+        """
+        _fs_args = copy.deepcopy(fs_args) or {}
+        _fs_open_args_load = _fs_args.pop("open_args_load", {})
+        _fs_open_args_save = _fs_args.pop("open_args_save", {})
+        _credentials = copy.deepcopy(credentials) or {}
+
+        protocol, path = get_protocol_and_path(filepath, version)
+
+        self._protocol = protocol
+        self._fs = fsspec.filesystem(self._protocol, **_credentials, **_fs_args)
+
+        super().__init__(
+            filepath=PurePosixPath(path),
+            version=version,
+            exists_function=self._fs.exists,
+            glob_function=self._fs.glob,
+        )
+
+        # Handle default load and save arguments
+        self._load_args = copy.deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        _fs_open_args_load.setdefault("mode", "r")
```

Comment:
Any specific reason why you're using `r` mode here? We've recently updated a lot of our datasets to just use the default modes from fsspec, so if that works for this dataset as well I'd suggest removing this line.

Reply:
sure, i believe this will work dropping the line. I will try it out and take it down.

There was no good reason apriory- it was inherited from somewhere else, i believe :)

Reply:
ok, the changed have been pushed

Reply:
Resolving, as there has been no feedback since 6 days